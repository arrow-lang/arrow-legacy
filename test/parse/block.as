// Some block statements
{x;}
{+30;}
{}

// Some block expressions
let x = {+{30;};};
