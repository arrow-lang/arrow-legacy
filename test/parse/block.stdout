{
  "tag": "Module",
  "span": "test/parse/block.as:1.1-7.19",
  "name": "block",
  "block": {
    "tag": "Block",
    "span": "test/parse/block.as:1.1-7.19",
    "statements": [
      {
        "tag": "Block",
        "span": "test/parse/block.as:2.1-5",
        "statements": [
          {
            "tag": "Identifier",
            "span": "test/parse/block.as:2.2-3",
            "text": "x"
          }
        ]
      },
      {
        "tag": "Block",
        "span": "test/parse/block.as:3.1-7",
        "statements": [
          {
            "tag": "Identity",
            "span": "test/parse/block.as:3.2-5",
            "operand": {
              "tag": "Integer",
              "span": "test/parse/block.as:3.3-5",
              "text": "30"
            }
          }
        ]
      },
      {
        "tag": "Block",
        "span": "test/parse/block.as:4.1-3",
        "statements": []
      },
      {
        "tag": "Slot",
        "span": "test/parse/block.as:7.1-19",
        "exported": false,
        "pattern": {
          "tag": "PatternIdentifier",
          "span": "test/parse/block.as:7.5-6",
          "text": "x",
          "mutable": false
        },
        "type": null,
        "initializer": {
          "tag": "Block",
          "span": "test/parse/block.as:7.9-18",
          "statements": [
            {
              "tag": "Identity",
              "span": "test/parse/block.as:7.10-16",
              "operand": {
                "tag": "Block",
                "span": "test/parse/block.as:7.11-16",
                "statements": [
                  {
                    "tag": "Integer",
                    "span": "test/parse/block.as:7.12-14",
                    "text": "30"
                  }
                ]
              }
            }
          ]
        }
      }
    ]
  }
}
